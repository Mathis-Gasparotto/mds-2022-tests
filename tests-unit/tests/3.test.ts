import axios from 'axios'
import { expect, test, vi } from 'vitest'
import { searchPokemon } from '../3'

vi.mock('axios')

const pokemons = {
  data: {
    results: [
      {
        name: 'Ouiiiii',
        url: 'oui.com'
      },
      {
        name: 'Non',
        url: 'non.oui'
      }
    ]
  }
}

test('search', async () => {
  await axios.get.mockResolvedValue(pokemons)
  const actual = await searchPokemon('Oui')

  expect(actual).toMatchSnapshot()
})

test('error', async () => {
  await axios.get.mockResolvedValue(pokemons)
  const actual = await searchPokemon('test')
  const expected = []

  expect(actual).toEqual(expected)
})
