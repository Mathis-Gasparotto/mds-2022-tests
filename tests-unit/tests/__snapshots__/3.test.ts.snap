// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`search 1`] = `
[
  {
    "name": "Ouiiiii",
    "url": "oui.com",
  },
]
`;
